scan/lib/scan/detect_values.rb

```diff
@@ -100,6 +109,63 @@ def self.filter_simulators(simulators, operator = :greater_than_or_equal, deploy
       end
     end
 
+    def self.default_os_version(os_type)
+      @os_versions ||= {}
+      @os_versions[os_type] ||= begin
+         UI.crash!("Unknown platform: #{os_type}") unless PLATFORMS.key?(os_type)
+         platform = PLATFORMS[os_type]
+
+         # list SDK version for currently running Xcode
+         sdks_output, = Open3.capture3('xcodebuild -showsdks -json')
+         sdk_version = begin
+           JSON.parse(sdks_output).find { |e| e['platform'] == platform[:simulator] }['sdkVersion']
+         rescue StandardError => e
+           UI.error(e)
+           UI.error("xcodebuild CLI broken, please run `xcodebuild` and make sure it works")
+           UI.user_error!("xcodebuild not working")
+         end
+
+         # Get runtime build from SDK version
+         runtime_output, = Open3.capture3('xcrun simctl runtime match list -j')
+         runtime_build = begin
+           JSON.parse(runtime_output).values.find { |elem| elem['platform'] == platform[:name] && elem['sdkVersion'] == sdk_version }['chosenRuntimeBuild']
+         rescue StandardError => e
```

Comment:
Same idea here
```suggestion
         runtime_build = begin
           runtime_output, _, status = Open3.capture3('xcrun simctl runtime match list -j')
           raise status unless status.success?
           JSON.parse(runtime_output).values.find { |elem| elem['platform'] == platform[:name] && elem['sdkVersion'] == sdk_version }['chosenRuntimeBuild']
         rescue StandardError => e
```

Reply:
This one is especially important because I think `match list` is only available starting Xcode 15, so your `simctl` wont have it if you don't have Xcodde 15 installed on your machine (and thus the latest `simctl` that comes with it installed in your system).

Sure, you check for that in `self.compatibility_constraint` before calling `default_os_version`, but `default_os_version` is a public method, so nothing ensures that this method won't be called by another method in the future (like someone creating a PR months from now to add some feature to fastlane and ending up using it without realizing it'd only work with Xcode 15).

I would instead recommend to move the check for the existence of the `match list` command from `self.compatibility_constraint` to this method instead—making it return `nil` if `match list` command does not exist—so that whoever calls it (even future PRs) gets the check.

Then in `self.compatibility_constaint`, always call `default = default_os_version(sim.os_type)` and decide which `->(name, sim)` lambda to use based on if `default` is `nil` or not. Or even simpler, now that `self.compatibility_constaint` method would be a bit trivial, just remove it and move its now-simplistic logic in `self.highest_compatible_simulator` directly, like this:
```ruby
    def self.highest_compatible_simulator(simulators, device_name)
      latest_os = default_os_version(sim.os_type)
      simulators
        .select { |sim| sim.name == name && (latest_os.nil? || Gem::Version.new(sim.os_version) <= latest_os) }
        .reverse
        .sort_by! { |sim| Gem::Version.new(sim.os_version) }
        .last
    end
```